Build and deploy proxy to server

jniemii opened the pull request:
Requires https://github.com/bytecraftoy/aalto-2022-infra/pull/10 to be merged first

Builds the Docker image in Actions, sends it to the server through ssh and enables&starts the systemd service.

jniemii committed:
use FQN for Dockerfile base image

jniemii committed:
enable --now instead of start for service

jniemii committed:
Dockerfile build and deploy ci

jniemii committed:
artifact rentention days: 90 -> 7

jaakkoha-bytecraft commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,66 @@
+      - name: Add host to known_hosts
+        run: echo "|1|51Fg/gQzqNfhMkuBnxLBSWCrl+s=|TbUOHEmNxVDTGdkxvvlmlN761QM= ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBMV0unNHDl5cqDEDYe5ADTCgXMZxr02k5M4mcI9gFPtlrUiDjlE0UTZ8ejgPWhILtK3iETzNb8FD53dMn6Ck/vQ=" >> ~/.ssh/known_hosts
+
+      - name: Stop proxy service
```
I believe these could be combined to `docker load` and then `systemctl restart proxyservice`

jniemii replied:
Resolved by [fa30132](https://github.com/bytecraftoy/aalto-2022-proxy/pull/2/commits/fa30132693a9e1e4953b00a7c856cc735ae035f3)

jniemii committed:
Simplify deploy restarting procedure

It still needs 'systemctl enable' to make sure service is autimatically started
